cobc/codegen.c

```diff
@@ -13926,6 +13926,29 @@ codegen_internal (struct cb_program *prog, const int subsequent_call)
 		/* Switch to main storage file */
 		output_target = cb_storage_file;
 	}
+
+	/* Decimal constants */
+	{
+		struct literal_list* m;
+		int comment_gen = 0;
+		for (m = literal_cache; m; m = m->next) {
+			if (CB_TREE_CLASS (m->x) == CB_CLASS_NUMERIC
```

Comment:
While unrelated to the original issue... I think we should drop the class check here, that's up to whatever defines `make_decimal`.
Suggestion: consider moving the initialization of `m` to its declaration.

Reply:
Note there are two other loops whose prefix are similar in `cobc.c` at 12604 and 12812 ; should I remove the class check in those two too ?

Reply:
Thanks for being thorough! Everything that checks for `make_decimal` to be set should not test for the class of the literal any more.